### PR TITLE
Adding Test For InferenceService Zero Initial Scale

### PR DESCRIPTION
**Pull request by brettmthompson**
## Description

This test is added to validate the new zero initial scale functionality added for [RHOAIENG-18965](https://issues.redhat.com/browse/RHOAIENG-18965).

The test creates an ISVC with 0 min-replicas and validates that no pods are created to initialize the ISVC. After creating the ISVC with 0 replicas, we validate that the model can be queried successfully.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/model_serving/model_server/serverless/test_zero_initial_scale.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    "model_namespace, ovms_kserve_serving_runtime, ovms_kserve_inference_service",
+    [
+        pytest.param(
+            {"name": "serverless-initial-scale-zero"},
+            RunTimeConfigs.ONNX_OPSET13_RUNTIME_CONFIG,
+            {
+                "name": ModelFormat.ONNX,
+                "model-version": ModelVersion.OPSET13,
+                "model-dir": "test-dir",
+                "deployment-mode": KServeDeploymentType.SERVERLESS,
```
Comment (rnetser) [inline tests/model_serving/model_server/serverless/test_zero_initial_scale.py:35]: please use [`ONNX_SERVERLESS_INFERENCE_SERVICE_CONFIG`](https://github.com/opendatahub-io/opendatahub-tests/blob/main/tests/model_serving/model_server/serverless/constants.py#L10)

File: tests/model_serving/model_server/serverless/test_zero_initial_scale.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+class TestServerlessInitialScaleZero:
+    @pytest.mark.dependency(name="test_no_serverless_pods_created_for_zero_initial_scale")
+    def test_no_serverless_pods_created_for_zero_initial_scale(self, admin_client, ovms_kserve_inference_service):
+        """Verify no pods are created when inference service initial scale is zero, i.e. zero min-replicas requested."""
+        verify_no_inference_pods(client=admin_client, isvc=ovms_kserve_inference_service)
+
+    @pytest.mark.dependency(name="test_no_serverless_replicas_created_for_zero_initial_scale")
+    def test_no_serverless_replicas_created_for_zero_initial_scale(self, admin_client, ovms_kserve_inference_service):
+        """Verify replica count is zero when inference service initial scale is zero"""
+        for deployment in Deployment.get(
```
Comment (rnetser) [inline tests/model_serving/model_server/serverless/test_zero_initial_scale.py:51]: iiuc [wait_for_replicas_in_deployment](https://github.com/opendatahub-io/opendatahub-tests/blob/main/utilities/infra.py#L140) can be used here

**Review by rnetser — CHANGES_REQUESTED** [review]

File: tests/model_serving/model_server/serverless/test_zero_initial_scale.py (hunk 2 above)
Comment (brettmthompson) [reply tests/model_serving/model_server/serverless/test_zero_initial_scale.py:51]: @rnetser  I'm not sure if we want to "wait" for the replicas in this scenario. Prior to the changes done for [RHOAIENG-18965](https://issues.redhat.com/browse/RHOAIENG-18965)., the deployment would be created with 1 replica then scaled down to 0 after some time, which runs the risk of still passing this test if it is scaled down prior to the timeout. What we want to test for here is that the deployment is created with 0 replicas immediately. Let me know if you still feel the `wait_for_replicas_in_deployment` function should be used here.

Comment (rnetser) [reply tests/model_serving/model_server/serverless/test_zero_initial_scale.py:51]: ok, got it.

File: tests/model_serving/model_server/serverless/test_zero_initial_scale.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            client=admin_client,
+            namespace=ovms_kserve_inference_service.namespace,
```
Comment (rnetser) [inline tests/model_serving/model_server/serverless/test_zero_initial_scale.py:53]: please use `label_selector`

```
Deployment.get(
            label_selector="xxx",
            client=admin_client,
            namespace=ovms_kserve_inference_service.namespace,
        )
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: tests/model_serving/model_server/serverless/test_zero_initial_scale.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    @pytest.mark.dependency(name="test_no_serverless_replicas_created_for_zero_initial_scale")
+    def test_no_serverless_replicas_created_for_zero_initial_scale(
+        self, admin_client, ovms_kserve_inference_service, ovms_kserve_serving_runtime
+    ):
+        """Verify replica count is zero when inference service initial scale is zero"""
+        labels = [
+            "serving.knative.dev/configurationGeneration=1",
+            create_isvc_label_selector_str(
+                ovms_kserve_inference_service, "depoyment", ovms_kserve_serving_runtime.name
+            ),
```
Comment (rnetser) [inline tests/model_serving/model_server/serverless/test_zero_initial_scale.py:55]: please use `arg=val` (validate_inference_output will enforce it)

Comment (brettmthompson) [reply tests/model_serving/model_server/serverless/test_zero_initial_scale.py:55]: @rnetser, I'm not sure I fully follow here. The [create_isvc_label_selector_str](https://github.com/opendatahub-io/opendatahub-tests/blob/main/utilities/general.py#L143) method returns a label in the `arg=val` format

Comment (rnetser) [reply tests/model_serving/model_server/serverless/test_zero_initial_scale.py:55]: sorry, i meant call the function `create_isvc_label_selector_str` with arg=val (isvc=ovms_kserve_inference_service  ...)

please reabse and run `pre-commit run --all-files`

**Review by rnetser — DISMISSED** [review]

**Review by adolfo-ab — APPROVED** [review]

**Review by dbasunag — DISMISSED** [review]

**Review by dbasunag — APPROVED** [review]

**Review by rnetser — APPROVED** [review]
